amr-wind/wind_energy/ABLAnelastic.cpp

```diff
@@ -0,0 +1,71 @@
+#include "amr-wind/wind_energy/ABL.H"
+
+namespace amr_wind {
+
+ABLAnelastic::ABLAnelastic(CFDSim& sim) : m_sim(sim)
+{
+    {
+        amrex::ParmParse pp("ABL");
+        pp.query("anelastic", m_is_anelastic);
+    }
+    {
+        amrex::ParmParse pp("incflo");
+        pp.queryarr("gravity", m_gravity);
+        pp.query("density", m_rho0_const);
+    }
+    if (m_is_anelastic) {
+        {
+            // ensure use of perturbational pressure form
+            amrex::ParmParse pp("ICNS");
+            pp.add("use_perturb_pressure", (bool)true);
+        }
+        m_sim.repo().declare_field("reference_density", 1, 0, 1);
+        m_sim.repo().declare_nd_field("reference_pressure", 1, 0, 1);
+    }
+}
+
+void ABLAnelastic::post_init_actions()
+{
+    if (!m_is_anelastic) {
+        return;
+    }
+    initialize_data();
+}
+
+void ABLAnelastic::post_regrid_actions()
+{
+    if (!m_is_anelastic) {
+        return;
+    }
+    m_sim.repo().declare_field("reference_density", 1, 0, 1);
+    m_sim.repo().declare_nd_field("reference_pressure", 1, 0, 1);
```

Comment:
these fields shouldn't need to be re-declared in post_regrid, right? They would just need to be recalculated